openff/bespokefit/executor/services/fragmenter/worker.py

```diff
@@ -67,8 +69,49 @@ def fragment(
                 "version": openff.bespokefit.__version__,
             },
         )
-        return result.json()
+        return _deduplicate_fragments(result).json()
 
     else:
         # no fragmentation and no mock fragments
         return "null"
+
+
+def _deduplicate_fragments(
+    fragmentation_result: FragmentationResult,
+) -> FragmentationResult:
+    """Remove symmetry equivalent fragments from the results."""
+    from collections import defaultdict
+
+    # group fragments which are the same
+    fragments_by_smiles = defaultdict(list)
+    for fragment_result in fragmentation_result.fragments:
+        fragments_by_smiles[
+            fragment_result.molecule.to_smiles(explicit_hydrogens=False)
+        ].append(fragment_result)
+
+    unique_fragments = []
+    # remove duplicated symmetry equivalent fragments
+    for fragments in fragments_by_smiles.values():
+        if len(fragments) == 1:
+            unique_fragments.extend(fragments)
+
+        else:
+            symmetry_groups = set()
```

Comment:
```suggestion
        if len(fragments) == 1:
            unique_fragments.extend(fragments)
            continue

        symmetry_groups = set()
```
As a very pedantic thing you could probably add a continue to the if statement and then remove / de-indent the else statement!